syntax = "proto3";
package cpp_service;

// Request messages

message SnapshotDataA
{
    uint32 id = 1;
}

message SubscribeDataB
{
    uint32 id = 1;
}

message Request
{
    oneof msg
    {
        SnapshotDataA snapshotDataA = 1;
        SubscribeDataB subscribeDataB = 2;
    }
}


// Response messages

message DataA
{
    uint32 id = 1;
    int32 value = 2;
}

message DataB
{
    uint32 id = 1;
    int64 ivalue = 2;
    string svalue = 3;
}
